packages/remix-dev/vite/plugin.ts

```diff
@@ -1419,8 +1411,9 @@ export const remixVitePlugin: RemixVitePlugin = (remixUserConfig = {}) => {
           let movedAssetPaths: string[] = [];
           for (let ssrAssetPath of ssrAssetPaths) {
             let src = path.join(serverBuildDirectory, ssrAssetPath);
-            if (!clientFilePaths.has(ssrAssetPath)) {
-              let dest = path.join(clientBuildDirectory, ssrAssetPath);
+            let dest = path.join(clientBuildDirectory, ssrAssetPath);
+
+            if (!fse.existsSync(dest)) {
```

Comment:
Since (assuming no errors) we were guaranteed to either move the SSR asset or delete it depending on whether it's present in the client assets, it's much safer to use the file system as the source of truth rather than trying to infer what's on disk based on the Vite manifest.